components/infobars/core/BUILD.gn

```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2023 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+static_library("core") {
+  sources = [
+    "brave_confirm_infobar_delegate.cc",
+    "brave_confirm_infobar_delegate.h",
+  ]
+
+  deps = [ "//components/infobars/core" ]
```

Comment:
I'd go with `public_deps` here, because `brave_confirm_infobar_delegate.h` uses a header from `//components/infobars/core`, and this header should have all generated headers ready to be used (mojom, strings, resources, etc.).

this is a subtle issue which in the current state may make a build flaky when some target decides to depend solely on `//brave/components/infobars/core`.

Reply:
Make sense. Updated. 👍🏼